### PR TITLE
Refactor MTE-1885 Move scripts and other files to firefox folder

### PR DESCRIPTION
**Pull request by isabelrios**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/MTE-1885)

Let's move more files into firefox-ios folder.

- L10n tests and import scripts
- Docs

Also let's rename the github actions mobile test eng own.

## :pencil: Checklist
You have to check all boxes before merging
- [ ] Filled in the above information (tickets numbers and description of your work)
- [ ] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed I updated documentation / comments for complex code and public methods

### CODE REVIEW

File: firefox-ios/l10n_comments.txt
Comment (isabelrios) [inline firefox-ios/l10n_comments.txt:1]: Not sure if we can even remove this file... it has not been updated in two years

File: firefox-ios/import-strings.sh
Comment (isabelrios) [inline firefox-ios/import-strings.sh:1]: @flodolo as commented on focus-ios PR, I'm also moving this script here. I have changed the path in the bootstrap file where it is executed. Locally I get the import working, but please let me know if I'm missing any other changes that could break the automation around this.

File: .github/workflows/firefox-ios-import-strings.yml
Comment (isabelrios) [inline .github/workflows/firefox-ios-import-strings.yml:1]: @flodolo I also changed the name for the focus  github action and seems that this change is not affecting anything, but doublechecking

Comment (flodolo) [reply .github/workflows/firefox-ios-import-strings.yml:1]: No risk, we have our own [workflow](https://github.com/mozilla-l10n/firefoxios-l10n/blob/main/.github/workflows/import_strings.yml), we only use scripts from this repo.

File: firefox-ios/l10n_comments.txt
Comment (flodolo) [reply firefox-ios/l10n_comments.txt:1]: @Delphine do you know what this is used for?

**Review by flodolo — COMMENTED** [review]
We should be good, since we use [LocalizationTools ](https://github.com/mozilla-l10n/firefoxios-l10n/blob/main/.github/workflows/import_strings.yml#L61)to do the actual extraction

File: firefox-ios/l10n_comments.txt
Comment (Delphine) [reply firefox-ios/l10n_comments.txt:1]: No idea. @boek Do you remember this?

File: .github/workflows/firefox-ios-glean-probe-scraper.yml
Comment (isabelrios) [inline .github/workflows/firefox-ios-glean-probe-scraper.yml:1]: @relud is it fine if we rename this?

File: .github/workflows/firefox-ios-publish-docc.yml
Comment (isabelrios) [inline .github/workflows/firefox-ios-publish-docc.yml:1]: @lmarceau checking with you if we can rename as author of the github action

File: .github/workflows/firefox-ios-stale.yml
Comment (isabelrios) [inline .github/workflows/firefox-ios-stale.yml:1]: @lmarceau checking with you if we can rename as author of the github action

File: .github/workflows/firefox-ios-update-nimbus-experiments.yml
Comment (isabelrios) [inline .github/workflows/firefox-ios-update-nimbus-experiments.yml:1]: @jeddai checking with you if we can rename without breaking anything as author of the github action

**Comment by flodolo** [comment]
And that's because the file is used from code that lives elsewhere, and needs to be updated as well
https://github.com/mozilla-mobile/LocalizationTools/blob/main/Sources/LocalizationTools/tasks/ExportTask.swift#L91-L98

**Comment by isabelrios** [comment]
> And that's because the file is used from code that lives elsewhere, and needs to be updated as well https://github.com/mozilla-mobile/LocalizationTools/blob/main/Sources/LocalizationTools/tasks/ExportTask.swift#L91-L98

Thanks for checking, I will not remove the `l10n_comments.txt` file then

File: firefox-ios/l10n_comments.txt
Comment (lmarceau) [reply firefox-ios/l10n_comments.txt:1]: It shouldn't be removed, it's used for the `WidgetIntent.strings` we have. Example [here](https://github.com/mozilla-mobile/firefox-ios/blob/main/WidgetKit/af.lproj/WidgetIntents.strings). The name definitely could be better than `l10n_comments` thought haha. But if the filename is changed then we should change it too in the [script](https://github.com/mozilla-mobile/LocalizationTools/blob/8d86df1f0cec47297adc0eba287e2b9a6774738b/Sources/LocalizationTools/tasks/ExportTask.swift#L91)

File: .github/workflows/firefox-ios-publish-docc.yml
Comment (lmarceau) [reply .github/workflows/firefox-ios-publish-docc.yml:1]: Yes should be safe to rename 👍

File: .github/workflows/firefox-ios-stale.yml
Comment (lmarceau) [reply .github/workflows/firefox-ios-stale.yml:1]: Yes should be safe to rename 👍

File: firefox-ios/l10n_comments.txt
Comment (isabelrios) [reply firefox-ios/l10n_comments.txt:1]: oh I see, thanks @lmarceau ... should I move it then or wait for another round of changes? Not sure I can create a PR in LocalizationTools repo or to handle that change there

Comment (lmarceau) [reply firefox-ios/l10n_comments.txt:1]: I think I can create a PR in there. The strings for v120 are currently being exported, so if we merge the current PR (17310) after that's done to avoid breaking l10n flow would be ok I think (should be done by end of week for sure) 👍 I just created a [task](https://mozilla-hub.atlassian.net/browse/FXIOS-7762) for myself to adjust the script in LocalizationTools once this PR is merged.

Comment (isabelrios) [reply firefox-ios/l10n_comments.txt:1]: that sounds great :) thanks!

File: .github/workflows/firefox-ios-glean-probe-scraper.yml
Comment (relud) [reply .github/workflows/firefox-ios-glean-probe-scraper.yml:1]: yes

**Comment by isabelrios** [comment]
I had to create a firefox-ios-tests folder inside firefox-ios and move the tests there. The way I created firefox-ios folder before would not allow me to move app folders in it in the future.

This PR is ready for review. Thanks for your comments so far.

**Review by lmarceau — APPROVED** [review]

**Comment by isabelrios** [comment]
@flodolo I'm moving the import-strings.sh script into firefor-ios folder. I have changed the bootstrap script to run it using the new location. Are there more changes needed so that the import/export keeps working? The import is working for me locally, but double checking as I don't want to break things

**Comment by flodolo** [comment]
> @flodolo I'm moving the import-strings.sh script into firefor-ios folder. I have changed the bootstrap script to run it using the new location. Are there more changes needed so that the import/export keeps working? The import is working for me locally, but double checking as I don't want to break things

We only use the bootstrap from this repo and LocalizationTools on our side. Once this is merged, you should verify that the import action works correctly in this repo.

**Comment by isabelrios** [comment]
> > @flodolo I'm moving the import-strings.sh script into firefor-ios folder. I have changed the bootstrap script to run it using the new location. Are there more changes needed so that the import/export keeps working? The import is working for me locally, but double checking as I don't want to break things
> 
> We only use the bootstrap from this repo and LocalizationTools on our side. Once this is merged, you should verify that the import action works correctly in this repo.

I will verify that for sure, thank you

**Comment by isabelrios** [comment]
FYi, This PR will not land until the v121 branch is cut from main on Monday to disrupt the less possible

**Comment by isabelrios** [comment]
> 😬 Would Sync Integration Tests run? I can give it a try.

In theory yes, on MacStadium I will fix the path to the test results once this lands

**Comment by clarmso** [comment]
> > 😬 Would Sync Integration Tests run? I can give it a try.
> 
> In theory yes, on MacStadium I will fix the path to the test results once this lands

🎉 I just confirmed that the `SyncIntegrationTests` could be run.

**Review by clarmso — APPROVED** [review]

**Comment by clarmso** [comment]
> > > 😬 Would Sync Integration Tests run? I can give it a try.
> > 
> > 
> > In theory yes, on MacStadium I will fix the path to the test results once this lands
> 
> 🎉 I just confirmed that the `SyncIntegrationTests` could be run.

I take it back. The TPS logs shows the following errors. @isabelrios Could you please update the paths to files needed by TPS?

```
2023-11-17 10:24:40.966 CROSSWEAVE ERROR: [phase -1] _executeTestPhase failed - Error opening input stream (invalid filename?): file:///Users/cso/workspace/firefox-ios/test_bookmark_desktop.js No traceback available
```

**Comment by isabelrios** [comment]
> > > > 😬 Would Sync Integration Tests run? I can give it a try.
> > > 
> > > 
> > > In theory yes, on MacStadium I will fix the path to the test results once this lands
> > 
> > 
> > 🎉 I just confirmed that the `SyncIntegrationTests` could be run.
> 
> I take it back. The TPS logs shows the following errors. @isabelrios Could you please update the paths to files needed by TPS?
> 
> ```
> 2023-11-17 10:24:40.966 CROSSWEAVE ERROR: [phase -1] _executeTestPhase failed - Error opening input stream (invalid filename?): file:///Users/cso/workspace/firefox-ios/test_bookmark_desktop.js No traceback available
> ```

@clarmso I have a copy of my branch on Jenkins and the tests passed there. It may be something on your virtual env that is chached? 
You can check on MacStadium: /Users/administrator/irios-tmp-test/irios-fork/firefox-ios/firefox-ios/firefox-ios-tests/Tests/SyncIntegrationTests
I have left the terminal open with my repo and branch. I commented the test_integration file to just run one test to try

**Comment by clarmso** [comment]
> > > > > 😬 Would Sync Integration Tests run? I can give it a try.
> > > > 
> > > > 
> > > > In theory yes, on MacStadium I will fix the path to the test results once this lands
> > > 
> > > 
> > > 🎉 I just confirmed that the `SyncIntegrationTests` could be run.
> > 
> > 
> > I take it back. The TPS logs shows the following errors. @isabelrios Could you please update the paths to files needed by TPS?
> > ```
> > 2023-11-17 10:24:40.966 CROSSWEAVE ERROR: [phase -1] _executeTestPhase failed - Error opening input stream (invalid filename?): file:///Users/cso/workspace/firefox-ios/test_bookmark_desktop.js No traceback available
> > ```
> 
> @clarmso I have a copy of my branch on MacStadium and the tests passed there. It may be something on your virtual env that is chached? You can check on MacStadium: /Users/administrator/irios-tmp-test/irios-fork/firefox-ios/firefox-ios/firefox-ios-tests/Tests/SyncIntegrationTests I have left the terminal open with my repo and branch. I commented the test_integration file to just run one test to try

😓 `pipenv clean` is needed in the directory after checking out the branch.

**Review by clarmso — APPROVED** [review]

**Review by OrlaM — APPROVED** [review]

**Pull request merged**
